sdks/unreal/Agones/Source/Agones/Private/AgonesComponent.cpp

```diff
@@ -146,6 +154,99 @@ void UAgonesComponent::GameServer(const FGameServerDelegate SuccessDelegate, con
 	Request->ProcessRequest();
 }
 
+void UAgonesComponent::EnsureWebSocketConnection()
+{
+    if (WatchWebSocket == nullptr)
+    {
+        if (!FModuleManager::LoadModulePtr<FWebSocketsModule>(TEXT("WebSockets")))
+        {
+            return;
+        }
+
+        TMap<FString, FString> Headers;
+
+        // Make up a WebSocket-Key value. It can be anything!
+        Headers.Add(TEXT("Sec-WebSocket-Key"), FGuid::NewGuid().ToString(EGuidFormats::Short));
+        Headers.Add(TEXT("Sec-WebSocket-Version"), TEXT("13"));
+        Headers.Add(TEXT("User-Agent"), TEXT("X-UnrealEngine-Agent"));
+
+        // Unreal WebSockets are not able to do DNS resolution for localhost for some reason
+        // so this is using the IPv4 Loopback Address instead.
+        WatchWebSocket = FWebSocketsModule::Get().CreateWebSocket(
+            FString::Format(TEXT("ws://127.0.0.1:{0}/watch/gameserver"), {*HttpPort}), TEXT(""));
+
+        WatchWebSocket->OnRawMessage().AddUObject(this, &UAgonesComponent::HandleWatchMessage);
+    }
+
+    if (WatchWebSocket != nullptr)
+    {
+        if (!WatchWebSocket->IsConnected())
+        {
+            WatchWebSocket->Connect();
+        }
+
+        // Only start the timer if there is a websocket to check.
+        if (!EnsureWebSocketTimerHandler.IsValid())
+        {
+            FTimerDelegate TimerDel;
+            TimerDel.BindUObject(this, &UAgonesComponent::EnsureWebSocketConnection);
+            GetWorld()->GetTimerManager().SetTimer(
+                EnsureWebSocketTimerHandler, TimerDel, HealthRateSeconds <= 0 ? 15 : HealthRateSeconds, true);

```

Comment:
would we want this configurable? Much like `HealthRateSeconds` we could have `WebSocketReconnectSeconds` or similar?

Reply:
No. In practice, this _never_ needs to reconnect. If it did, the agent itself had to restart, and if you want to guarantee up-to-date data, the correct answer is to use the normal GameServer(SuccessDel, ErrorDel) method to make direct request. If anything, this should be fixed at a 15-30 second second timer. It has nothing to do with health and just about making sure that the connection is mostly up.